src/codeu/chat/server/Server.java

```diff
@@ -66,13 +76,90 @@
   private final Relay relay;
   private Uuid lastSeen = Uuid.NULL;
 
+  private void loadState() {
```

Comment:
Consider encapsulating this logic in a helper class (TransactionFileReader?). Then, simply pass the controller instance to the reader. That way, you can have this logic clearly separate out elsewhere. This would also make this logic more unit testable :)